src/Kernel/TableKernel.cc

```diff
@@ -184,9 +208,9 @@ TableKernel<Dimension>::TableKernel(const KernelType& kernel,
   mNumPoints(numPoints),
   mMinNperh(std::max(minNperh, 1.1/kernel.kernelExtent())),
   mMaxNperh(maxNperh),
-  mInterp(0.0, kernel.kernelExtent(), numPoints,      [&](const double x) { return kernel(x, 1.0); }),
-  mGradInterp(0.0, kernel.kernelExtent(), numPoints,  [&](const double x) { return kernel.grad(x, 1.0); }),
-  mGrad2Interp(0.0, kernel.kernelExtent(), numPoints, [&](const double x) { return kernel.grad2(x, 1.0); }),
+  mInterp(0.0, kernel.kernelExtent(), numPoints, Wlookup<KernelType>(kernel)),
+  mGradInterp(0.0, kernel.kernelExtent(), numPoints, gradWlookup<KernelType>(kernel)),
+  mGrad2Interp(0.0, kernel.kernelExtent(), numPoints, grad2Wlookup<KernelType>(kernel)),
```

Comment:
Ah I see, so you have to go back to externally defined functors.  This is harmless enough, but kind of a bummer since I like the compact (and I think clearer) lambda function version.  Do we think this is something that newer versions of nvcc can/will correct?  Seems like they're not supporting standard C++ things here.

Also, since RAJA relies on lamda functions for it's trickery as well, what's teh salient difference here?  Is it the reference capturing [&] version?

Reply:
RAJA requires capturing by value [=] so a copy of the object must be taken into a host-device lambda.

In this specific case the lambda is used for construction of the Interpreter. When VVI + Cuda are enabled the underlying ManagedSharedPtr tries to invoke the constructor on both the host and device. Meaning `kernel` isn't available to the lambda invoked on the device, we need a copy of it to be passed over to the device in order to be able to call it correctly. 

There are a couple of ways of getting back to an interface where we provide a lambda:
1. Construct interpreters from factory functions that are able to construct w/ appropriate size on Host + Device, then fill the interpreter on the host and copy the contents to the device.
2. Have the interprter take a handle to the kernel object then define the lambda as `[=](const double, const kernel) { return kernel ... }` so the interpreter can pass the kernel internally in it's constructor.